cmd/internal/cli/common_linux.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/opencontainers/runtime-tools/generate"
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/config"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/config/oci"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+	"github.com/sylabs/singularity/internal/pkg/util/exec"
+	"github.com/sylabs/singularity/internal/pkg/util/fs"
+	"github.com/sylabs/singularity/pkg/image"
+	"github.com/sylabs/singularity/pkg/image/unpacker"
+	singularityConfig "github.com/sylabs/singularity/pkg/runtime/engines/singularity/config"
+)
+
+// singularityConfPath holds the path to singularity.conf file
+var singularityConfPath = filepath.Join(buildcfg.SYSCONFDIR, "singularity/singularity.conf")
+
+// convertSquashfsToSandbox take a single image file in SIF or squashfs format
+// to convert it to a sandbox image and returns the path to the sandbox.
+func convertSquashfsToSandbox(filename string, unsquashfsPath string) (string, error) {
+	img, err := image.Init(filename, false)
+	if err != nil {
+		return "", fmt.Errorf("could not open image %s: %s", filename, err)
+	}
+	defer img.File.Close()
+
+	if !img.HasRootFs() {
+		return "", fmt.Errorf("no root filesystem found in %s", filename)
+	}
+
+	// squashfs only
+	if img.Partitions[0].Type != image.SQUASHFS {
+		return "", fmt.Errorf("not a squashfs root filesystem")
+	}
+
+	// create a reader for rootfs partition
+	reader, err := image.NewPartitionReader(img, "", 0)
+	if err != nil {
+		return "", fmt.Errorf("could not extract root filesystem: %s", err)
+	}
+	s := unpacker.NewSquashfs()
+	if !s.HasUnsquashfs() && unsquashfsPath != "" {
+		s.UnsquashfsPath = unsquashfsPath
+	}
+
+	// keep compatibility with v2
+	tmpdir := os.Getenv("SINGULARITY_LOCALCACHEDIR")
+	if tmpdir == "" {
+		tmpdir = os.Getenv("SINGULARITY_CACHEDIR")
+	}
+
+	// create temporary sandbox
+	dir, err := ioutil.TempDir(tmpdir, "rootfs-")
+	if err != nil {
+		return "", fmt.Errorf("could not create temporary sandbox: %s", err)
+	}
+
+	// extract root filesystem
+	if err := s.ExtractAll(reader, dir); err != nil {
+		os.RemoveAll(dir)
+		return "", fmt.Errorf("root filesystem extraction failed: %s", err)
+	}
+
+	return dir, err
+}
+
+// getfileConfig returns singularity configuration read from singularity.conf
+// file.
+func getFileConfig() (*singularityConfig.FileConfig, error) {
```

Comment:
this looks like something that should be part of `pkg/runtime/engines/singularity/config`, no?